### PR TITLE
Pcf8574 improvements

### PR DESCRIPTION
**Pull request by barbudor**
## Description:

Improvements on PCF8574

Following Discussion https://github.com/arendst/Tasmota/discussions/11176 with @turin400:

- Up to now, PCF inputs were useless. Allows usage of PCF inputs through 4 optional USE_PCF8574_xxx build flags:
```
      #define USE_PCF8574_SENSOR                   // enable PCF8574 inputs and outputs state in SENSOR message
      #define USE_PCF8574_DISPLAYINPUT             // enable PCF8574 inputs in Web page
      #define USE_PCF8574_MQTTINPUT                // enable MQTT message on input change detection : stat/%topic%/PCF8574_INP = {"Time":"2021-03-07T16:19:23+01:00","PCF8574-1_INP":{"D1":1}}
      #define USE_PCF8574_EVENTINPUT               // enable Event generation on input change detection :  stat/%topic%/EVENT PCF8574-1_D1=1
```
Found during the dev of above:
- Outputs : fix out-of-syncsync between chip pin's state and Tasmota power state
```
  If save_data=OFF and no_power_feedback=OFF
      Power state is synced from reading the chip state
  else
      Insure chip pin is properly synced on power-on
```

Comments welcome on 
- How I performed the sync trying to mimic `SetPowerOnState()` https://github.com/arendst/Tasmota/blob/8630debb7f9d73a7dcb42903f1474f44e71e1fe3/tasmota/support_tasmota.ino#L332-L338
- Inputs messages and events syntax
- Should we keep all 4 USE_PCF8574_xxx ? (will have to add final ones in `my_user_config.h` for reference)

**Related issue (if applicable):** fixes #<Tasmota issue number goes here>

Discussion https://github.com/arendst/Tasmota/discussions/11176 

## Checklist:
  - [X] The pull request is done against the latest development branch
  - [X] Only relevant files were touched
  - [X] Only one feature/fix was added per PR and the code change compiles without warnings
  - [X] The code change is tested and works on Tasmota core ESP8266 V.2.7.4.9
  - [ ] The code change is tested and works with core ESP32 V.1.0.5
  - [X] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

**Comment by turin400** [comment]
Thank you. It works great, the rules work and it's simple.
It was pending on the pcf8574 support.

**Comment by arendst** [comment]
Can't we live without the EVENTINPUT? SHouldn't the MQTTINPUT enough to handle rules?

Other than that I think it looks fine.

File: tasmota/xdrv_28_pcf8574.ino [hunk 1]
```diff
@@ -188,6 +207,68 @@ void HandlePcf8574(void)
...
+      if (input != last_input) { // don't scan bits if no change (EVERY_50_MS !)
+        for (uint8_t pin = 0 ; pin < 8 ; ++pin) {
+          if (bitRead(input_mask,pin) && bitRead(input,pin) != bitRead(last_input,pin)) {
+  #ifdef USE_PCF8574_MQTTINPUT
+            ResponseTime_P(PSTR(",\"PCF8574%c%d_INP\":{\"D%i\":%i}}"), IndexSeparator(), idx +1, pin, bitRead(input,pin));
+            MqttPublishPrefixTopic_P(RESULT_OR_STAT, PSTR("PCF8574_INP"));
+            if (Settings.flag3.hass_tele_on_power) {  // SetOption59 - Send tele/%topic%/SENSOR in addition to stat/%topic%/RESULT
+                MqttPublishSensor();
+            }
+  #endif // #ifdef USE_PCF8574_MQTTINPUT
```
Comment (arendst) [inline tasmota/xdrv_28_pcf8574.ino:258]: I also prefer these messages to be in one message to solve MQTT flooding.

**Comment by barbudor** [comment]
Hi Theo @arendst 
Thanks for the review
I just picked-up inspiration from what is done in the MCP driver that is doing both.
I'll remove the EVENT.
I'll squash all that into a new clean PR

**Comment by barbudor** [comment]
Final PR : https://github.com/arendst/Tasmota/pull/11301

**Pull request closed without merging**